appimageinstall.sh

```diff
@@ -0,0 +1,174 @@
+    awk -v home="$HOME" -v app_name="$app_name" '
+    BEGIN { in_action = 0 }
+    /^\[Desktop Action/ { in_action = 1 }
+    /^Exec=/ {
+        if (in_action) {
+            split($0, parts, "=")
+            sub(/^[^ ]+/, "", parts[2])  # Remove the first word (original command)
+            print "Exec=" home "/.local/share/AppImage/" app_name ".AppImage" parts[2]
+        } else {
+            print "Exec=" home "/.local/share/AppImage/" app_name ".AppImage"
+        }
+        next
+    }
+    /^Icon=/ { print "Icon=" home "/.local/share/icons/" app_name ".png"; next }
+    { print }
+    ' "${desktop_file}" > "${desktop_file}.tmp" && mv "${desktop_file}.tmp" "${desktop_file}"
+
+    # Clean up extracted files
+    rm -rf squashfs-root
+}
+
+uninstall_appimage() {
+    local app_name="$1"
+
+    # Remove AppImage
+    rm -f ~/.local/share/AppImage/${app_name}.AppImage
+
+    # Remove .desktop file
+    rm -f ~/.local/share/applications/${app_name}.desktop
+
+    # Remove icon
+    rm -f ~/.local/share/icons/${app_name}.png
+
+    echo "(︶︹︺) Uninstalled ${app_name}"
+}
+
+choose_appimage_version() {
+    echo "(◕‿◕✿) Please choose the version to install:"
+    echo "+------------------------------------------------+"
+    echo "| 1) Optimized - Blazing fast and compatible     |"
+    echo "|    with modern devices                         |"
+    echo "+------------------------------------------------+"
+    echo "| 2) Generic - Slow but compatible with older    |"
+    echo "|    devices                                     |"
+    echo "+------------------------------------------------+"
+    echo "| 3) Exit                                        |"
+    echo "+------------------------------------------------+"
+    read -p "Enter your choice (1/2/3): " version_choice
+
+    case $version_choice in
+        1)
+            url=$(curl -s https://api.github.com/repos/zen-browser/desktop/releases/latest | grep "browser_download_url.*zen-specific.AppImage\"" | cut -d '"' -f 4)
+            download_until_success "$url" ~/Downloads/ZenBrowser.AppImage
+            process_appimage ~/Downloads/ZenBrowser.AppImage ZenBrowser
+            ;;
+        2)
+            url=$(curl -s https://api.github.com/repos/zen-browser/desktop/releases/latest | grep "browser_download_url.*zen-generic.AppImage\"" | cut -d '"' -f 4)
+            download_until_success "$url" ~/Downloads/ZenBrowser.AppImage
+            process_appimage ~/Downloads/ZenBrowser.AppImage ZenBrowser
+            ;;
+        3)
+            echo "(⌒‿⌒) Exiting..."
+            exit 0
+            ;;
+        *)
+            echo "(•ˋ _ ˊ•) Invalid choice. Exiting..."
+            exit 1
+            ;;
+    esac
+}
+
+main_menu() {
+    echo "(★^O^★) What would you like to do?"
+    echo "+------------------------------------------------+"
+    echo "| 1) Install                                     |"
+    echo "+------------------------------------------------+"
```

Comment:
What about an option to update? We have zsync appimages

Reply:
I could do that, but let's me test it out, never try to update with a zsync before.